### PR TITLE
7973-add-ngc-prefix

### LINKED ISSUE #7973: Enhance `_download_from_ngc` to support no-prefix or different prefix data
So far, it's alway use the default prefix:
```
def _download_from_ngc(
    download_path: Path, filename: str, version: str, remove_prefix: str | None, progress: bool
) -> None:
    # ensure prefix is contained
    filename = _add_ngc_prefix(filename)
...
```
We can make use of `_add_ngc_prefix`'s `prefix` arg

### PR DESCRIPTION
**Pull request by yiheng-wang-nv**
Fixes #7973 .

### Description

A few sentences describing the changes proposed in this pull request.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/bundle/scripts.py [hunk 1]
```diff
@@ -231,10 +236,16 @@ def _download_from_ngc(
 
 
 def _download_from_ngc_private(
-    download_path: Path, filename: str, version: str, remove_prefix: str | None, repo: str, headers: dict | None = None
+    download_path: Path,
+    filename: str,
+    version: str,
+    repo: str,
+    prefix: str = "monai_",
+    remove_prefix: str | None = "monai_",
```
Comment (KumoLiu) [inline monai/bundle/scripts.py:244]: Look like the change of the default value here will make the behavior not consistent with previous.
If no special reason, I would suggest revert it back.
cc @mingxin-zheng

Comment (yiheng-wang-nv) [reply monai/bundle/scripts.py:244]: no, the default is always monai_. Please check: https://github.com/Project-MONAI/MONAI/blob/23ade147fd9a3c32d754b28c2ae4c365b924b04a/monai/bundle/scripts.py#L440

Comment (KumoLiu) [reply monai/bundle/scripts.py:244]: My bad, closed as resolved. Thanks.

File: monai/bundle/scripts.py [hunk 2]
```diff
@@ -217,10 +217,15 @@ def _remove_ngc_prefix(name: str, prefix: str = "monai_") -> str:
 
 
 def _download_from_ngc(
-    download_path: Path, filename: str, version: str, remove_prefix: str | None, progress: bool
+    download_path: Path,
+    filename: str,
+    version: str,
+    prefix: str = "monai_",
```
Comment (KumoLiu) [inline monai/bundle/scripts.py:223]: Do we also need add this arg in the download and pass it to this one?

Comment (KumoLiu) [reply monai/bundle/scripts.py:223]: Discussed this offline with @yiheng-wang-nv 
Will leave this one for this PR. If we have a case need this arg in the download and load, will enhance them later.

**Review by Nic-Ma — APPROVED** [review]

**Pull request merged**